x-pack/platform/packages/shared/kbn-dissect-heuristics/src/types.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * A node in the Dissect pattern AST
+ */
+export type DissectASTNode = DissectFieldNode | DissectLiteralNode;
+
+/**
+ * Represents a field in the pattern AST
+ */
+export interface DissectFieldNode {
+  type: 'field';
+  name: string;
+  modifiers?: {
+    rightPadding?: boolean;
+    skip?: boolean;
+    namedSkip?: boolean;
```

Comment:
Isn't `namedSkip = true` just a node with a `name` and `skip = true`? What would `{ skip: false, namedSkip: true }` mean?

Are all of these modifiers combinable and valid? For example can you skip a field and also append it to another one? (`%{+?field_1->}`)